src/form_components/repeating_group.coffee

```diff
@@ -0,0 +1,155 @@
+FormRenderer.Models.ResponseFieldRepeatingGroup = FormRenderer.Models.BaseFormComponent.extend
+  group: true
+  field_type: 'repeating_group'
+
+  initialize: ->
+    FormRenderer.Models.BaseFormComponent::initialize.apply @, arguments
+    @entries = []
+
+  validateComponent: ->
+    for entry in @entries
+      entry.formComponents.invoke('validateComponent')
+
+  setExistingValue: (entryValues) ->
+    if @isRequired()
+      # If the field is required, ensure that there is at least one value present
+      if !entryValues || entryValues.length == 0
+        entryValues = [{}]
+
+    else # Field is optional...
+      # If entryValues is not set, then add an empty value.
+      if !entryValues
+        entryValues = [{}]
+      # If entryValues is an empty array, the field is skipped.
+      else if _.isArray(entryValues) && _.isEmpty(entryValues)
+        @set('skipped', true)
+
+    @entries = _.map entryValues, (value) =>
+      new FormRenderer.Models.ResponseFieldRepeatingGroupEntry({ value }, @fr, @)
+
+  addEntry: ->
+    @entries.push(
+      new FormRenderer.Models.ResponseFieldRepeatingGroupEntry({}, @fr, @)
+    )
+
+  removeEntry: (idx) ->
+    @entries.splice(idx, 1)
+
+    if @entries.length == 0
+      @set('skipped', true)
+
+  isSkipped: ->
+    !!@get('skipped')
+
+  getValue: ->
+    if @isSkipped()
+      []
+    else
+      _.invoke @entries, 'getValue'
+
+  maxEntries: ->
+    if @get('maxentries')
+      parseInt(@get('maxentries'), 10) || Infinity
+    else
+      Infinity
+
+  canAdd: ->
+    @entries.length < @maxEntries()
+
+FormRenderer.Models.ResponseFieldRepeatingGroupEntry = Backbone.Model.extend
+  initialize: (_attrs, @fr, @repeatingGroup) ->
+    @initFormComponents @repeatingGroup.get('children'), @get('value') || {}
+
+  reflectConditions: ->
+    @view.reflectConditions()
+
+  canRemove: ->
+    @repeatingGroup.entries.length > 1
```

Comment:
Yeah, that's the intent here. Rather than just removing all repeated sections within to signal they don't have an answer, we have the user "skip" the question. That was the original design, and I think it makes intuitive sense when you consider a question that asks "How many firearms do you want to declare?" -- I would think a user with no firearms would be more likely to "skip this question" than remove all fields. Happy to have a discussion if you feel differently though; I'm not glued to the decision.

Reply:
I'm cool with that